app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -392,6 +394,19 @@ public void onWindowVideoAvailabilityChanged(@NonNull WindowWidget aWindow) {
             mWhatsNewWidget.getPlacement().parentHandle = mWindows.getFocusedWindow().getHandle();
             mWhatsNewWidget.show(UIWidget.REQUEST_FOCUS);
         }
+
+        EngineProvider.INSTANCE.loadExtensions()
+                .thenAcceptAsync(aVoid -> {
+                    Log.d(LOGTAG, "WebExtensions loaded");
+                    mWindows.restoreSessions();
+                }, getServicesProvider().getExecutors().mainThread())
+                .exceptionally(throwable -> {
+                    String msg = throwable.getLocalizedMessage();
+                    if (msg != null) {
+                        Log.e(LOGTAG, "Extensions load error: " + msg);
+                    }
```

Comment:
Do we not want to try and restore sessions even if there is an error loading the extensions.

Reply:
I've updated to restore sessions even if the extensions load fails. I guess it would be nice to also notify the user, otherwise we might be loading sessions which won't work properly due the the lack of extensions. I'll open a follow up, we might need a less intrusive notification system than an alert for these cases, ideally a window bottom-up timed sliding notification.